python/cuml/linear_model/linear_regression.pyx

```diff
@@ -267,9 +261,8 @@ class LinearRegression(Base):
                           "column currently.", UserWarning)
             self.algo = 0
 
-        self.coef_ = cudf.Series(zeros(self.n_cols,
-                                       dtype=self.dtype))
-        cdef uintptr_t coef_ptr = get_cudf_column_ptr(self.coef_)
+        self._coef_ = CumlArray.zeros(self.n_cols, dtype=self.dtype)
+        cdef uintptr_t coef_ptr = self._coef_.ptr
```

Comment:
Not a change for this PR specifically, but more for the future... stylistically I wonder if in these cases it would be better to just keep the _coef_ around and use `_coef_.ptr` at the point where we actually have to pointerize it. I.e. the convention would be "use CumlArrays as locals". My argument there would be that it then gets easier to reason about local types since they'll all be cuml arrays.

Reply:
I think that's a good idea